rpc/marshalling.go

```diff
@@ -0,0 +1,121 @@
+package rpc // import "collectd.org/rpc"
+
+import (
+	"collectd.org/api"
+	pb "collectd.org/rpc/proto/types"
+	"github.com/golang/protobuf/ptypes"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+// MarshalValue converts an api.Value to a pb.Value.
+func MarshalValue(v api.Value) (*pb.Value, error) {
+	switch v := v.(type) {
+	case api.Counter:
+		return &pb.Value{
+			Value: &pb.Value_Counter{Counter: uint64(v)},
+		}, nil
+	case api.Derive:
+		return &pb.Value{
+			Value: &pb.Value_Derive{Derive: int64(v)},
+		}, nil
+	case api.Gauge:
+		return &pb.Value{
+			Value: &pb.Value_Gauge{Gauge: float64(v)},
+		}, nil
+	default:
+		return nil, grpc.Errorf(codes.InvalidArgument, "%T values are not supported", v)
+	}
+}
+
+// UnmarshalValue converts a pb.Value to an api.Value.
+func UnmarshalValue(in *pb.Value) (api.Value, error) {
+	switch pbValue := in.GetValue().(type) {
```

Comment:
There's a lof of pbValue and pb.Value in this code, making it hard to read imho. I'd use `v` for this variable instead.